packages/rspack/tests/configCases/builtins/provide/index.js

```diff
@@ -0,0 +1,68 @@
+it("should provide a module for a simple free var", function () {
+	expect(aaa).toBe("aaa");
+});
+
+it("should provide a module for a nested var", function () {
+	expect(bbb.ccc).toBe("bbbccc");
+	var x = bbb.ccc;
+	expect(x).toBe("bbbccc");
+});
+
+// it("should provide a module for a nested var within a IIFE's argument", function () {
+// 	(function (process) {
+// 		expect(process.env.NODE_ENV).toBe("development");
+// 		var x = process.env.NODE_ENV;
+// 		expect(x).toBe("development");
+// 	})(process);
+// });
+
```

Comment:
The commented out tests are failing as of now. Those cases needs to be handled / transformed properly.